src/realtime-callbacks.js

```diff
@@ -56,6 +56,80 @@ module.exports.setNow = function(f) {
 };
 let _now = Date.now;
 
+// Internal hash of active debouncers
+const _debouncers = {};
+
+class Debouncer {
+
+    constructor(jobName) {
+        this.jobName = jobName;
+    }
+
+    complete() {
+        delete _debouncers[this.jobName];
+        this.func();
+    }
+
+    start(func, waitMs) {
+        if (func !== undefined) this.func = func;
+        if (waitMs !== undefined) this.waitMs = waitMs;
+        this.timeoutId = setTimeout(this.complete.bind(this), this.waitMs);
+    }
+
+    stop() {
+        if (this.timeoutId) {
+            clearTimeout(this.timeoutId);
+            delete this.timeoutId;
+        }
+    }
+}
+
+/**
+ * Implementation of debounce a which will call the callback after
+ * a waitMs period if it is not debounced within that period.
+ *
+ * After a jobName has been defined it may be called without
+ * {func} or {waitMs} arguments to debounce with initial settings.
+ * New {func} or {waitMs} params will over-write ones previously
+ * set in the debouncer.
+ * If called without a jobName a random one is selected.
+ *
+ * @param {function} func   callback to be called after a delay
+ * @param {Number} waitMs  number of milliseconds to delay by
+ *
+ * @return {String} a jobName to reference the debouncer.
+ */
+
+module.exports.debounce = function(jobName, func, waitMs) {
+    if (jobName === undefined) {
+        while(_debouncers[jobName = Math.random().toString(16).slice(2, 10)]);
+    }
+    let debouncer = _debouncers[jobName];
+    if (debouncer) {
+        debouncer.stop();
+    } else {
+        _debouncers[jobName] = debouncer = new Debouncer(jobName, func, waitMs);
+    }
+    debouncer.start(func, waitMs);
+    return jobName;
+};
+/**
+ * Clear a debounce job..
+ *
+ * @param {String} a jobName to reference the debouncer.
+ *
+ * @return {Bool} whether a debouncer with jobName was cleared.
```

Comment:
`{boolean}` for primitive `booleans` please. I don't think JSDoc3 will like `{Bool}`.